### DISCUSSION TITLE
Automatic Font Size selection

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#3679] [Help]
Hello,

a few month ago I was looking for a pice ago which automacally makes a piece of text fitting into a given textarea:
Snibo gay me this peace of code, which worked great:
( ^
	      -size 4372x640 -font %FONT% -background None -fill #3b3734 caption:%2 -trim -resize 2336x320 -gravity center -extent 2336x320 ^
)^

However, it doesn't work if there are blanks at the beginning or end. 

Here is the error message:
magick: geometry does not contain image ` ' @ warning/attribute.c/GetImageBoundingBox/510.

Is there a possibilty to handle blanks like normal caracters?

Here is the full code. IM7, windows bat
----------------------------------
```
@echo off
set TEMPLATEDIR=C:\\Users\\Thomas\\Google Drive\\POD\\imagemagic\\templates\\V2\\IH32
set TEMPLATE="%TEMPLATEDIR%\\%3.png"
if a%3 =="a" goto :start
set TEMPLATE="%TEMPLATEDIR%\\F%3.png"


:start
echo %TEMPLATE%
set FONT="C:\\Users\\Thomas\\Google Drive\\POD\\imagemagic\\fonts\\MagnoliaScript.ttf"


magick (  %TEMPLATE% ^
          ( ^
		    %1-crop.jpg -gravity north -crop 2199:3088+0+0 +repage -resize 2220x3110! ^
		  ) ^
		  -gravity NorthWest -geometry +4125+720   -composite  ^
	   ) ^
	   ( ^
	      -size 4372x640 -font %FONT% -background None -fill #3b3734 caption:%2 -trim -resize 2336x320 -gravity center -extent 2336x320 ^
	    )^
		-gravity northwest -geometry +4050+4048 -composite -background white -gravity Center -extent 7056x4704 %1-32-full.jpg
```

![luna-32-poster](https://user-images.githubusercontent.com/48103248/118239441-1548fc00-b49a-11eb-9d7e-13e5ae7751ff.jpg)

**Comment by snibgo** [comment]
What version of IM?

If your caption contains spaces, it needs to be quoted. If it has leading spaces, insert a backslash \ at the start.

If that doesn't fix the problem please show what command you use to run the script, and show all the input images.

**Comment by tbmuc** [comment]
IM-Version:  ImageMagick 7.0.10-40 with windows cmd shell

I have  changed caption:%2 to caption:"%2".

The script is called 

fullih20 file "Luna 2006-     " 
or
fullih20 file "Luna 2006-\ \ \ \ "

The error message disappeared, but the blanks at the  end are still ignored.

**Comment by snibgo** [comment]
You have:
```
caption:%2 -trim 
```
If the caption has leading or trailing spaces, "-trim" will remove them.

**Comment by tbmuc** [comment]
hm, without trim the mechanism to find the optimal font size seems not to work. And the text is placed on the left instead in the center of the area.

**Reply by fmw42** [reply]
Use -gravity center before the caption:
